python/pyiceberg/table/metadata.py

```diff
@@ -159,10 +163,14 @@ def construct_refs(cls, data: Dict[str, Any]):
     remove oldest metadata log entries and keep a fixed-size log of the most
     recent entries after a commit."""
 
-    sort_orders: List[Dict[str, Any]] = Field(alias="sort-orders", default_factory=list)
+    sort_orders: List[SortOrder] = Field(alias="sort-orders", default_factory=list)
     """A list of sort orders, stored as full sort order objects."""
 
-    default_sort_order_id: int = Field(alias="default-sort-order-id", default=DEFAULT_SORT_ORDER_UNSORTED)
+    @property
+    def bound_sort_orders(self) -> List[SortOrder]:
+        return [sort_order.bind(self.current_schema()) for sort_order in self.sort_orders]
```

Comment:
For now, I'd probably just make one accessor that binds the default sort order to the current schema. I'm not sure if it will be valuable to have all of the known sort orders bound to a particular schema.